feat(webapps): improve process definition page

danielkelemen opened the pull request:
related to camunda/camunda-bpm-platform#3612

tasso94 reviewed (commented):
I haven't reviewed everything in detail yet but played a little bit around with the UI. First of all: great job! Looks very good and makes handling high amounts of process definitions much easier 🙂 👍 

Some things I noticed during playing around:
* ❌ Other search pill based widgets have a default filter that is applied when entering a value and hitting return; that's currently not the case but we should probably make it consistent with other places.
![8098c88e-0db0-47dc-859f-8573340d4b0f](https://github.com/camunda/camunda-bpm-platform/assets/3015690/d9bfcbbc-a545-4a5c-8f07-517d00a6df10)
* ❌ Saving a query via the floppy disk button doesn't work for me. 
![image](https://github.com/camunda/camunda-bpm-platform/assets/3015690/b24b487d-6dc7-43eb-8c81-e34ee2a9b98f)
* 🔧 Now that we can filter for process definitions, should we add copy widgets to process definition name and key for easier copy & paste?

tasso94 reviewed (changes requested):
Looks good to me. Just some minor hints:

When no process definitions are deployed, the section toggle button overlays the search widget: ![Screenshot 2023-09-19 at 09 57 18](https://github.com/camunda/camunda-bpm-platform/assets/3015690/cf1bd198-22bc-4fb6-a088-629458ca4908)

danielkelemen commented:
> Other search pill based widgets have a default filter that is applied when entering a value and hitting return; that's currently not the case but we should probably make it consistent with other places.

Nice catch, I didn't notice this feature yet.

> Saving a query via the floppy disk button doesn't work for me.

Fixed.

> Now that we can filter for process definitions, should we add copy widgets to process definition name and key for easier copy & paste?

Good idea! Added it.

tasso94 commented:
Fixed a small problem with the copy widget when the definition name is empty.

tasso94 reviewed (approved):
👍
